src/services/auth.service.ts

```diff
@@ -0,0 +1,39 @@
+import { type DeepPartial } from 'typeorm'
+import { User } from '../entity/User'
+import AppDataSource from '../data-source'
+import * as bcrypt from 'bcrypt'
+
+class CustomError extends Error {
+  status: number
+  constructor (message: string, status: number) {
+    super(message)
+    this.status = status
+    this.name = this.constructor.name
+  }
+}
+
+export async function signUp (data: DeepPartial<User>): Promise<User> {
+  const user = AppDataSource.manager.create(User, data)
```

Comment:
check beforehand if the user already exist or add unique constraint on user email